sage/example-transition/src/asset/mod.rs

```diff
@@ -7,22 +7,24 @@ pub mod hero_jam;
 pub type AssetId = u32;
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Encode, Decode, MaxEncodedLen, TypeInfo)]
-pub enum AssetVariant<BlockNumber> {
-	HeroJam(hero_jam::HeroJamAsset<BlockNumber>),
+pub enum AssetVariant<BlockNumber, Balance> {
+	HeroJam(hero_jam::HeroJamAsset<BlockNumber, Balance>),
 }
 
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Encode, Decode, MaxEncodedLen, TypeInfo)]
-pub struct Asset<BlockNumber> {
-	pub asset_variant: AssetVariant<BlockNumber>,
+pub struct Asset<BlockNumber, Balance> {
+	pub asset_variant: AssetVariant<BlockNumber, Balance>,
```

Comment:
I think the generics are a hassle. To simplify the lives of downstream implementors, I think we should expose all the types of our runtime in ajuna primitives, so that they can use the actual concrete type in the implementation. The only downside to that is that the tests need to use the same types.

Reply:
I think exposing the types in ajuna-primitives may be a good idea. But for the example I would still use the generics.